File: doc-examples/example-java/src/test/java/example/MyHttpFunctionTest.java
package example;

import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpResponseMessage;
import io.micronaut.http.HttpStatus;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class MyHttpFunctionTest {

    @Test
    void routesRequestsToControllers() {
        try (MyHttpFunction function = new MyHttpFunction()) {
            HttpResponseMessage response = function.request(HttpMethod.GET, "/hello").invoke();

            assertEquals(HttpStatus.OK.getCode(), response.getStatusCode());
            assertEquals("Hello World", response.getBody());
        }
    }
}

File: doc-examples/example-kotlin/src/test/kotlin/example/MyHttpFunctionTest.kt
package example

import com.microsoft.azure.functions.HttpMethod
import io.micronaut.http.HttpStatus
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class MyHttpFunctionTest {

    @Test
    fun routesRequestsToControllers() {
        MyHttpFunction().use { function ->
            val response = function.request(HttpMethod.GET, "/hello").invoke()

            assertEquals(HttpStatus.OK.code, response.statusCode)
            assertEquals("Hello World", response.body)
        }
    }
}
